src/Components/Todo/TodoItem.tsx

```diff
@@ -0,0 +1,62 @@
+import classNames from 'classnames';
+import React, { useState } from 'react';
+import { Todo } from '../../types/Todo';
+
+type Props = {
+  todo: Todo;
+  onTodoRemove: (todoId: number) => void;
+  isRemovingCompleted: boolean;
+};
+
+export const TodoItem: React.FC<Props> = ({
+  todo,
+  onTodoRemove,
+  isRemovingCompleted,
+}) => {
+  const [isRemoving, setIsRemoving] = useState<boolean>(false);
+  const isNewTodo = todo.id === 0;
+
+  const isRemovingCompletedTodo = isRemovingCompleted && todo.completed;
```

Comment:
Instead of using 'isRemovingCompleted' prop, you can use 'isRemoving && todo.completed' to check if the component is removing a completed todo.

Reply:
This change is responsible for opening the loader for every removed todo after clicking on the button 'Clear completed', I tried your variant and it doesn't work